src/common/News/useFetchStockNews.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from 'react';
+import {
+  NEWS_API_URL,
+  NEWS_API_KEY,
+} from 'react-native-dotenv';
+
+export default (query, page = 1, pageSize = 20) => {
+  const [news, setNews] = useState([]);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const newsUrl = new URL(`${NEWS_API_URL}/everything`);
+  const parsedQuery = (typeof query.join) === 'function'
+    ? query.join(' OR ')
+    : query;
+  newsUrl.searchParams.append('apiKey', NEWS_API_KEY);
+  newsUrl.searchParams.append('q', parsedQuery);
+  newsUrl.searchParams.append('pageSize', pageSize);
+  newsUrl.searchParams.append('page', page);
+  newsUrl.searchParams.append('sortBy', 'publishedAt');
+  newsUrl.searchParams.append('language', 'en');
+
+  useEffect(() => {
+    setIsLoading(true);
+    const fetchData = async () => {
```

Comment:
do we really need to define this method within the whole hook and useEffect? We should have something like this
```js
async function fetchNews(newsUrl) {
  ...
}

export default (...) => {
  useEffect(() => {
    try {
      setIsLoading(true);
      const result = fetchNews();
      setNews(page === 1 ? result : [...news, ...result])
    } catch(err) {
      ...
    }
  })
}
```

Reply:
done